libs/cardano-ledger-api/src/Cardano/Ledger/Api/Scripts.hs

```diff
@@ -47,12 +71,159 @@ import Cardano.Ledger.Alonzo.Scripts (
   pattern RewardingPurpose,
   pattern SpendingPurpose,
  )
-import Cardano.Ledger.Api.Era ()
+import Cardano.Ledger.Api.Era
 import Cardano.Ledger.Api.Scripts.Data
+import Cardano.Ledger.Conway.Governance (ProposalProcedure, Voter)
 import Cardano.Ledger.Conway.Scripts (
   ConwayEraScript (..),
   pattern ProposingPurpose,
   pattern VotingPurpose,
  )
-import Cardano.Ledger.Core (EraScript (..), hashScript, isNativeScript, validateNativeScript)
+import Cardano.Ledger.Core (
+  EraScript (..),
+  TxCert,
+  hashScript,
+  isNativeScript,
+  validateNativeScript,
+ )
+import Cardano.Ledger.Dijkstra.Scripts (
+  DijkstraEraScript (..),
+  pattern GuardingPurpose,
+ )
 import Cardano.Ledger.Hashes (ScriptHash)
+import Cardano.Ledger.Mary.Value (PolicyID)
+import Cardano.Ledger.Plutus (Language)
+import Cardano.Ledger.TxIn (TxIn)
+import Data.Word (Word32)
+
+class EraScript era => AnyEraScript era where
+  anyEraMaxLanguage :: Maybe Language
+  default anyEraMaxLanguage ::
+    AlonzoEraScript era => Maybe Language
+  anyEraMaxLanguage = Just (eraMaxLanguage @era)
```

Comment:
Could we turn this into an associated type instead, like we do with `ProtVerHigh`?
```suggestion
  type EraLanguageHigh era :: Language
```

Reply:
It would be possible, however I do not think it would be any more useful than the regular value level `eraMaxLanguage`.
We already have a mechanism for dealing with language support at the type level through `EraPlutusTxInfo` et al.

If you have a use case in mind then we could certainly add it. Adding something just because we can is not a very good development strategy in general.

In either case this suggestion would be applicable to `AlonzoEraScript` type class, rather than here.

I'll resolve this conversation and if you find a use case for your suggestion then please open a ticket and we'll definitely add it.